tokio/src/sync/mpsc/unbounded.rs

```diff
@@ -223,6 +223,22 @@ impl<T> UnboundedReceiver<T> {
         self.chan.close();
     }
 
+    /// Checks if the channel has been closed. This happens when the
+    /// [`UnboundedReceiver::close`] method is called.
+    ///
+    /// [`UnboundedReceiver::close`]: crate::sync::mpsc::UnboundedReceiver::close
+    ///
+    /// ```
+    /// let (tx, mut rx) = tokio::sync::mpsc::unbounded_channel::<()>();
+    /// assert!(!rx.is_closed());
+    ///
+    /// rx.close();
+    /// assert!(rx.is_closed());
+    /// ```
+    pub fn is_closed(&self) -> bool {
+        self.chan.is_closed()
```

Comment:
As discussed on Discord, this method should only return `true` if the channel is also empty.